### PR TITLE
APM-292194: Run Logs POC in container

### PR DESCRIPTION
**Pull request by pgutkowski**
Initial POC of log forwarder in container. `dynatrace_client` has been copied from Azure log forwarder solution. Job queue and Pub/sub subscriber parameters have been selected arbitrarily, we need to run performance tests to fine tune them. Default behavior of `run_docker` scripts has been preserved.

I've introduced simple integration test with wiremock, as it happened to be simpler than messing with mocks. New stage has been added to `.travis.yml` to run the tests.

### CODE REVIEW

File: src/lib/logs/log_forwarder.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+PROJECT_ID = os.environ.get('LOGS_SUBSCRIPTION_PROJECT', None)
+SUBSCRIPTION_ID = os.environ.get('LOGS_SUBSCRIPTION_ID', None)
+EXECUTION_PERIOD_SECONDS = get_int_environment_value("DYNATRACE_LOG_INGEST_SENDING_WORKER_EXECUTION_PERIOD", 60)
+
+
+def run_logs(logging_context: LoggingContext):
+    if not PROJECT_ID or not SUBSCRIPTION_ID:
+        raise Exception("Cannot start pubsub streaming pull - LOGS_SUBSCRIPTION_PROJECT or LOGS_SUBSCRIPTION_ID are not defined")
+    # Settings for job queue size and subscriber should be fine tuned, but we have to do performance tests first
+    job_queue = Queue(10_000)
```
Comment (sq2gxo) [inline src/lib/logs/log_forwarder.py:37]: this could be put into constant

**Review by sq2gxo — APPROVED** [review]

**Review by Dawid-Kaszubski — APPROVED** [review]

File: src/lib/logs/log_forwarder.py (hunk 1 above)
Comment (pgutkowski) [reply src/lib/logs/log_forwarder.py]: Fixed

**Review by mswiatkowska — APPROVED** [review]

File: src/lib/logs/log_forwarder.py [hunk 2]
```diff
@@ -21,33 +21,36 @@
...
+from lib.logs.logs_processor import create_process_message_handler, SENDING_WORKER_EXECUTION_PERIOD_SECONDS
 from lib.logs.logs_sending_worker import create_log_sending_worker_loop
 
 PROJECT_ID = os.environ.get('LOGS_SUBSCRIPTION_PROJECT', None)
 SUBSCRIPTION_ID = os.environ.get('LOGS_SUBSCRIPTION_ID', None)
-EXECUTION_PERIOD_SECONDS = get_int_environment_value("DYNATRACE_LOG_INGEST_SENDING_WORKER_EXECUTION_PERIOD", 60)
+
+
+MAX_MESSAGES_PROCESSED = 10_000
+MAX_WORKERS = 10_000
```
Comment (sq2gxo) [inline src/lib/logs/log_forwarder.py:33]: should it be 100?

**Review by sq2gxo — CHANGES_REQUESTED** [review]

File: src/lib/logs/log_forwarder.py (hunk 2 above)
Comment (pgutkowski) [reply src/lib/logs/log_forwarder.py]: Good catch, thank you! Fixed

**Comment by pgutkowski** [comment]
@sq2gxo you've asked whether the streaming pull will require additional open ports - according to official docs, there are no additional ports opened, just server replying with long-existing stream of GRPC messages.

**Pull request merged**